parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -539,7 +540,12 @@ private static final ParquetMetadata readFooter(InputFile file, ParquetReadOptio
       throw new RuntimeException("corrupted file: the footer index is not within the file: " + footerIndex);
     }
     f.seek(footerIndex);
-    return converter.readParquetMetadata(f, options.getMetadataFilter());
+    ByteBuffer footerBytesBuffer = ByteBuffer.allocate(footerLength);
```

Comment:
Could you add some comments about the purpose of this code part?

Reply:
I think this is used to read all footer into a byte buffer instead of reading them byte by byte.

Reply:
Sorry to miss this for so long... Will add a comment later.

Reply:
Comment added